docs/DeveloperGuide.md

```diff
@@ -1,29 +1,211 @@
+
+* `PlayerProfile`: Represents the player's profile and is responsible for storing and updating the player's information.
+* `Parser`: Responsible for parsing the user input.
+* `ResponseManager`: Responsible for generating the response to the user input.
+* `EconoCraftLogic`: Responsible for executing user command and updating the game state.
+* `MiniGame`: Responsible for handling the mini-games that the player can play to gain rewards.
+* `CommandFactory`: Responsible for using the parsed user input to produce executable commands.
+
+#### Component interaction:
+
+The *Sequence Diagram* below showcases the interaction between the different components of the EconoCraft Pro 
+application when a user inputs a command `work`.
+
+For `CommandFactory` and `Minigame`,
+* Each defines its API for creating commands and mini-games respectively, where
+  `Command` is the API for `CommandFactory` and `MiniGame` is the API for `MiniGame`.
+* Implements its functionality with concrete classes such as `WorkCommand` and `TypingGame`.
+
+The sections below would give more details of each component.
+
+## UI components
+
+UI consists of the following components:
+* `Parser`
+* `ResponseManager`
+* `CommandType`
+
+![UI.png](UML%20diagram%2FUI.png)
+
+The **UI components**,
+* `Parser` parses the user input for `CommandFactory` to produce `Command`
+* `ResponseManager` generate the response to the user according to the *command execution* and *game logic*.
+
+## EconoCraftLogic components
+
+Here is the partial class diagram of the `EconoCraftLogic` component:
+
+![Logic.png](UML%20diagram%2FLogic.png)
+
+> [!NOTE]
+> - The `XYZ` in `XYZCommand` represents the exact command e.g., `WorkCommand`, `RestCommand`, `ExerciseCommand`.
+> - The `XYZ` in `XYZMiniGame` represents the exact mini-game e.g., `TypingGame`, `TicTacToe`, etc.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+Here is the sequence diagram of the `EconoCraftLogic` executing the command:
+
+![Work.png](UML%20diagram%2FWork.png)
+
+The `EconoCraftLogic` mechanism:
+
+1. `EconoCraftLogic` receives the user input string and pass it into `CommandFactory`.
+2. `CommandFactory` would use `Parser` to parse the user input string and produce a `Command`.
+3. `EconoCraftLogic` would execute the `Command` which will:
+   * update the player profile accordingly.
+   * use `ResponseManager` to generate the response to the user.
+
+## MiniGame components
+
+Here is the partial class diagram of the `MiniGame` component:
+
+![MiniGame.png](UML%20diagram%2FMiniGame.png)
+
+The `MiniGame` mechanism:
+1. For command `work`, `rest`, and `exercise`, they would have their respective mini-games.
+2. When these commands are generated and executed in `EconoCraftLogic`, the respective mini-game would be played.
+3. The command would then update the player profile according to the mini-game result.
+
+# Implementation
+
+## MiniGame - Typing Game
```

Comment:
The different functionalities of the different mini-games is comprehensively but sensibly written about in the DG. Good job.

Reply:
Thank you! Will take note of the issues:)